packages/synapse-react-client/src/components/CardDeck/TableQueryCardDeck.tsx

```diff
@@ -0,0 +1,137 @@
+    headerImageFileHandleColumnName,
+    cardDeckType,
+    linkConfig,
+  } = props
+  const entity = useGetEntity(entityId)
+  const queryBundleRequest: QueryBundleRequest = {
+    concreteType: 'org.sagebionetworks.repo.model.table.QueryBundleRequest',
+    entityId,
+    partMask:
+      SynapseConstants.BUNDLE_MASK_QUERY_SELECT_COLUMNS |
+      SynapseConstants.BUNDLE_MASK_QUERY_RESULTS,
+    query: {
+      sql: `select * from ${entityId}`,
+    },
+  }
+  const { data: queryResultBundle } =
+    useGetQueryResultBundle(queryBundleRequest)
+
+  const titleColumnIndex = getFieldIndex(titleColumnName, queryResultBundle)
+  const descriptionColumnIndex = getFieldIndex(
+    descriptionColumnName,
+    queryResultBundle,
+  )
+  const ctaButtonTextColumnIndex = getFieldIndex(
+    ctaButtonTextColumnName,
+    queryResultBundle,
+  )
+  const ctaButtonURLColumnIndex = getFieldIndex(
+    ctaButtonURLColumnName,
+    queryResultBundle,
+  )
+  const titleIconColumnIndex = getFieldIndex(
+    titleIconFileHandleColumnName,
+    queryResultBundle,
+  )
+  const headerImageColumnIndex = getFieldIndex(
+    headerImageFileHandleColumnName,
+    queryResultBundle,
+  )
+
+  const cards = queryResultBundle?.queryResult!.queryResults.rows.map(
+    (el): CardDeckCardProps => {
+      const values = el.values as string[]
+      if (values.some(value => value === null)) {
+        // We cast values above assuming there are no null values, emit a warning just in case.
+        console.warn('Row has null value(s) when no nulls expected')
+      }
+
+      const title = values[titleColumnIndex]
+      const description = values[descriptionColumnIndex]
+      const ctaButtonText = values[ctaButtonTextColumnIndex]
+      const ctaButtonURL = values[ctaButtonURLColumnIndex] ?? ''
+      const titleIconFileHandleIdValue = values[titleIconColumnIndex] ?? ''
+      const headerImageFileHandleIdValue = values[headerImageColumnIndex] ?? ''
+
+      const schema: Record<string, number> = {}
+      queryResultBundle?.queryResult!.queryResults.headers.forEach(
+        (header, colIndex) => {
+          schema[header.name] = colIndex
+        },
+      )
+      const { href } = getLinkParams(ctaButtonURL, linkConfig, values, schema)
+
+      const titleIconFileHandleAssociation = getFileHandleAssociation(
+        entity.data,
+        titleIconFileHandleIdValue,
+        `syn${el.rowId}`,
+      )
+
+      const titleIcon = titleIconFileHandleAssociation ? (
+        <ImageFileHandle
+          fileHandleAssociation={titleIconFileHandleAssociation}
+        />
+      ) : undefined
+
+      const headerImageFileHandleAssociation = getFileHandleAssociation(
+        entity.data,
+        headerImageFileHandleIdValue,
+        `syn${el.rowId}`,
+      )
+
+      const headerImage = headerImageFileHandleAssociation ? (
+        <ImageFileHandle
+          fileHandleAssociation={headerImageFileHandleAssociation}
+        />
+      ) : undefined
```

Comment:
Create the `ImageFileHandle` elements here so we can just pass them into the cards as arbitrary `ReactNode`s